implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/v2/DynatraceExporterV2.java

```diff
@@ -0,0 +1,362 @@
+        return toGauge(meter);
+    }
+
+    private void throwIfValueIsInvalid(Double value) {
+        if (value.isNaN()) {
+            throw new IllegalArgumentException("Value cannot be NaN.");
+        }
+        if (value.isInfinite()) {
+            throw new IllegalArgumentException("Value cannot be infinite.");
+        }
+    }
+
+    Stream<String> toGauge(Meter meter) {
+        return streamOf(meter.measure()).map(
+                measurement -> {
+                    try {
+                        throwIfValueIsInvalid(measurement.getValue());
+                        String metricKey = createMetricKey(meter.getId().getName(),
+                                measurement.getStatistic().getTagValueRepresentation());
+                        if (metricKey == null) {
+                            return null;
+                        }
+                        return createMetricBuilder(meter, metricKey)
+                                .setDoubleGaugeValue(measurement.getValue())
+                                .serialize();
+                    } catch (MetricException e) {
+                        logger.warn(String.format("Could not serialize metric with name %s: %s", meter.getId().getName(), e.getMessage()));
+                    } catch (IllegalArgumentException iae) {
+                        logger.warn(String.format("Illegal value for metric with name %s: %s Dropping...", meter.getId().getName(), iae.getMessage()));
+                    }
+                    return null;
+                })
+                .filter(Objects::nonNull);
+    }
+
+    Stream<String> toCounter(Meter meter) {
+        return streamOf(meter.measure()).map(
+                measurement -> {
+                    try {
+                        throwIfValueIsInvalid(measurement.getValue());
+                        return createMetricBuilder(meter, meter.getId().getName())
+                                .setDoubleCounterValueDelta(measurement.getValue())
+                                .serialize();
+                    } catch (MetricException e) {
+                        logger.warn(String.format("Could not serialize metric with name %s: %s", meter.getId().getName(), e.getMessage()));
+                    } catch (IllegalArgumentException iae) {
+                        logger.warn(String.format("Illegal value for metric with name %s: %s Dropping...", meter.getId().getName(), iae.getMessage()));
+                    }
+                    return null;
+                })
+                .filter(Objects::nonNull);
+    }
+
+    private Metric.Builder createMetricBuilder(Meter meter, String metricKey) {
+        return metricBuilderFactory
+                .newMetricBuilder(metricKey)
+                .setDimensions(fromTags(meter.getId().getTags()))
+                .setTimestamp(Instant.ofEpochMilli(clock.wallTime()));
+    }
+
+    private String createMetricKey(String name, String tagValueRepresentation) {
```

Comment:
```suggestion
    private static String createMetricKey(String name, String tagValueRepresentation) {
```
It seems like this is only used for Gauges, maybe mention this in the name.